src/js/controllers/index.js

```diff
@@ -140,16 +140,19 @@ angular.module('copayApp.controllers').controller('indexController', function($r
       self.setCustomBWSFlag();
       if (fc.isPrivKeyExternal()) {
         self.needsBackup = false;
-        self.openWallet();
       } else {
-        storageService.getBackupFlag(self.walletId, function(err, val) {
-          if (!fc.credentials.mnemonic)
+        if (!fc.credentials.mnemonic) {
+          storageService.setBackupFlag(self.walletId, function(err) {
+            $log.debug('Backup stored');
             self.needsBackup = false;
-          else
+          });
+        } else {
+          storageService.getBackupFlag(self.walletId, function(err, val) {
             self.needsBackup = self.network == 'testnet' ? false : !val;
-          self.openWallet();
-        });
+          });
+        }
       }
+      self.openWallet();
```

Comment:
`openWallet()` will be opened before the other async functions. I 'm not sure it 's a good practice. `openWallet()` should be inside these async functions